ORM/DataObject.php

```diff
@@ -1682,8 +1700,7 @@ public function getRelationType($component) {
 	 * @throws InvalidArgumentException
 	 */
 	public function inferReciprocalComponent($remoteClass, $remoteRelation) {
-		/** @var DataObject $remote */
-		$remote = $remoteClass::singleton();
+		$remote = DataObject::singleton($remoteClass);
```

Comment:
Fair call; fine to leave for now. The lack of type hinting from our DI system can be a problem for another day ;-)